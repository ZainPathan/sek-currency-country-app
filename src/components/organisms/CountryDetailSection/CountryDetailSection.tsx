import React from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import IconComponent from '../../atoms/IconComponent';

import useCountryDetailsSectionStyle from './CountryDetailsSection.style';

import { labels } from '../../../core/constants';
import { GetAmountInCurrency } from '../../../core/utils';

interface CountryDetailSectionProps {
    countriesList: any,
    currMap: object,
    currencyAmountInSEK: number,
    loadingState: string
}

const CountryDetailSection: React.FC<CountryDetailSectionProps> = (props) => {
    const classes = useCountryDetailsSectionStyle();

    return (
        <>
            {props.countriesList && props.countriesList.length > 0 ?
                (
                    <TableContainer className={classes.tableContainer} component={Paper}>
                        <Table className={classes.table} aria-label="simple table">
                            <TableHead className={classes.tableHead}>
                                <TableRow>
                                    {labels.countryDetailsHeaders.map((detailHeader, index) => {
                                        return (
                                            <TableCell
                                                align={index > 1 ? 'right' : 'inherit'}
                                                key={detailHeader}
                                            >
                                                {detailHeader}
                                            </TableCell>);
                                    })}
                                </TableRow>
                            </TableHead>
                            <TableBody>
                                {props.countriesList.map((country) => (
                                    <TableRow key={country.name}>
                                        <TableCell>
                                            <IconComponent src={country.flag}/>
                                        </TableCell>
                                        <TableCell component="th" scope="row">
                                            {country.name}
                                        </TableCell>
                                        <TableCell align="right">{country.capital}</TableCell>
                                        <TableCell align="right">{country.population}</TableCell>
                                        <TableCell align="right">{country?.currencies[0]?.name}</TableCell>
                                        <TableCell align="right">
                                            {props.currMap && Object.keys(props.currMap).length > 0 &&
                                                props.currencyAmountInSEK ?
                                                    (GetAmountInCurrency(
                                                        props.currMap,
                                                        country?.currencies[0]?.code,
                                                        country?.currencies[0]?.symbol,
                                                        props.currencyAmountInSEK))
                                                    : null
                                            }</TableCell>
                                    </TableRow>
                                ))}
                            </TableBody>
                        </Table>
                    </TableContainer>
                )
                : ((props.loadingState === 'ERROR') ?
                    (
                        <div className={classes.errorContent}>
                            {labels.searchCountryError}
                        </div>
                    )
                    : null)
            }
        </>
    );
};

export default CountryDetailSection;