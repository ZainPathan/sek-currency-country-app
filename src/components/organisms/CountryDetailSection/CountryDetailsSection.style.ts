import { makeStyles } from '@material-ui/core/styles';

const useCountryDetailsSectionStyle = makeStyles({
    tableContainer: {
        margin: 12,
        width: 'inherit',
        border: '1px solid black'
    },
    table: {
        minWidth: 650,
    },
    tableHead: {
        '& .MuiTableCell-head': {
            fontWeight: 700,
        }
    },
    errorContent: {
        margin: 12,
        padding: 10,
        border: '1px solid black',
        color: 'red'
    }
});

export default useCountryDetailsSectionStyle;