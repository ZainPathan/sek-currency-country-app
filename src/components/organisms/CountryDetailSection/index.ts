import CountryDetailSection from './CountryDetailSection';

export {
    CountryDetailSection
};

export default CountryDetailSection;