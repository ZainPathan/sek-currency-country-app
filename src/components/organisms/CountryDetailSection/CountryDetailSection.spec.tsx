import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import CountryDetailSection from "./CountryDetailSection";

describe('CountryDetailsSection component tests', () => {
   const mockCountriesList = [
       {
           "name": "France",
           "capital": "Paris",
           "population": 66710000,
           "currencies": [
               {
                   "code": "EUR",
                   "name": "Euro",
                   "symbol": "€"
               }
           ],
           "flag": "https://restcountries.eu/data/fra.svg",
       },
       {
           "name": "French Polynesia",
           "capital": "Papeetē",
           "population": 271800,
           "currencies": [
               {
                   "code": "XPF",
                   "name": "CFP franc",
                   "symbol": "Fr"
               }
           ],
           "flag": "https://restcountries.eu/data/pyf.svg",
       }
   ];
    const currencyRatesMap = {
        "SEK": 10.097733,
        "EUR": 1,
        "XPF": 119.884226
    };
    const mockCurrencyAmountInSEK = 100;
    const loadingState = false;

    const mockCurrMap = Object.keys(currencyRatesMap).reduce((result, curr) => {
        result[curr] = currencyRatesMap[curr]/currencyRatesMap['SEK'];
        return result;
    }, {});

   it('snapshot test', () => {
       const tree = renderer
           .create(
                <CountryDetailSection
                    countriesList={mockCountriesList}
                    currMap={mockCurrMap}
                    currencyAmountInSEK={mockCurrencyAmountInSEK}
                    loadingState={loadingState}
                />
           )
           .toJSON();
       expect(tree).toMatchSnapshot();
   });

   it('renders with proper country name in list', () => {
       const testCountryText = 'France';
       const { getByText } = render(
           <CountryDetailSection
               countriesList={mockCountriesList}
               currMap={mockCurrMap}
               currencyAmountInSEK={mockCurrencyAmountInSEK}
               loadingState={loadingState}
           />
       );
       expect(getByText(testCountryText)).toBeTruthy();
   });

   it('renders with proper local currency amount', () => {
       const testLocalCurrencyAmount = '€ 9.90';
       const { getByText } = render(
           <CountryDetailSection
               countriesList={mockCountriesList}
               currMap={mockCurrMap}
               currencyAmountInSEK={mockCurrencyAmountInSEK}
               loadingState={loadingState}
           />
       );
       expect(getByText(testLocalCurrencyAmount)).toBeTruthy();
   });
});