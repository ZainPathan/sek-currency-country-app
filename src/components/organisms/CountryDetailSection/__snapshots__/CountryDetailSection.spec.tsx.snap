// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDetailsSection component tests snapshot test 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root makeStyles-tableContainer-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="simple table"
    className="MuiTable-root makeStyles-table-2"
    role={null}
  >
    <thead
      className="MuiTableHead-root makeStyles-tableHead-3"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Country Flag
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Country Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Capital
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Population
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Currency
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Amount in Local Currency
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root"
          >
            <img
              alt=""
              className="makeStyles-flag-5"
              src="https://restcountries.eu/data/fra.svg"
            />
          </span>
        </td>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          France
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          Paris
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          66710000
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          Euro
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
           € 9.90
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root"
          >
            <img
              alt=""
              className="makeStyles-flag-5"
              src="https://restcountries.eu/data/pyf.svg"
            />
          </span>
        </td>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          French Polynesia
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          Papeetē
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          271800
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          CFP franc
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
           Fr 1187.24
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
